### ISSUE TITLE
[Feature] Add more data for hypercrx

### ISSUE THREAD

**Issue by frank-zsy** [hypertrons/hypertrons-crx#395]
### Description

I would like to add more data into Hypercrx. The first part is top 1 million repos(by star) on GitHub history details.

The data looks like below:

``` json
{
	"id": "7691631",
	"n": "moby/moby",
	"ic": {
		"20151": 5681,
		"20152": 5417,
		"20153": 6493,
		"20154": 7416,
		"20155": 6403,
		"20156": 5764,
		...
	},
	"oi": {
		"20151": 278,
		"20152": 255,
		"20153": 367,
		"20154": 403,
		"20155": 284,
		"20156": 342,
		...
	},
	"op": {
		"20151": 338,
		"20152": 339,
		"20153": 514,
		"20154": 537,
		"20155": 422,
		"20156": 341,
		...
	},
	"pm": {
		"20151": 245,
		"20152": 237,
		"20153": 406,
		"20154": 403,
		"20155": 330,
		"20156": 255,
		...
	},
	"rc": {
		"20151": 935,
		"20152": 1027,
		"20153": 1384,
		"20154": 2054,
		"20155": 1755,
		"20156": 913,
		...
	},
	"ad": {
		"20151": 21750,
		"20152": 19467,
		"20153": 42833,
		"20154": 47983,
		"20155": 69026,
		"20156": 76092,
		 ...
	},
	"de": {
		"20151": 6519,
		"20152": 71321,
		"20153": 20958,
		"20154": 27393,
		"20155": 20630,
		"20156": 64424,
		...
	},
	"s": {
		"20151": 763,
		"20152": 743,
		"20153": 870,
		"20154": 901,
		"20155": 824,
		"20156": 918,
		...
	},
	"f": {
		"20151": 899,
		"20152": 743,
		"20153": 1089,
		"20154": 963,
		"20155": 875,
		"20156": 985,
		...
	},
	"p": {
		"20151": 279,
		"20152": 217,
		"20153": 606,
		"20154": 426,
		"20155": 375,
		"20156": 442,
		...
	}
}
```

The field means:

- ic: issue comment count
- oi: open issue count
- op: open pull count
- mp: merge pull count
- rc: review comment count
- ad: merged additions code lines
- de: merged deletions code lines
- s: stars
- f: forks

All the data is grouped by month and only the number greater than 0 included.

All the data will be uploaded onto OSS storage and open to access.

Can you gays put these data into Hypercrx in proper way?

**Comment by tyn1998** [comment]
Ok~

But we need to fix #382 first. It is troublesome.

**Comment by will-ww** [comment]
So great, and we can do more visualization work soon~

**Comment by tyn1998** [comment]
Hi, guys, I'm considering how to present these data.

I think there will be **two ways** to show the data. 

- One is show them as a whole such as a multiple lines chart:
  <img width ="300" src="https://user-images.githubusercontent.com/32434520/181477221-466fe71e-7a78-443a-a661-265a5de963dc.png"/>

- Another is present each data in different and detailed way, making them more embeded. For "fork" and "start", the embeded view may be like:
  <img width ="900" src="https://user-images.githubusercontent.com/32434520/181482571-bf4a8320-c78c-4ab3-b655-c5e2755ef226.png"/>

So in a big picture, each kind of data will be presented in two ways for different purposes. if any more detailed plan comes up, I will update it here. 

PS: This issue will be split to several issues so it will act as a "main issue" to track subtasks.

**Comment by wxharry** [comment]
One vote for the latter one

**Comment by tyn1998** [comment]
There is no need to choose. We want both 😄 . (In my view)

An overview is for comparing and capturing screenshot, while an embeded view can gives users more immersed experience.

**Comment by tyn1998** [comment]
Subtask:

- [ ] https://github.com/hypertrons/hypertrons-crx/issues/431

**Comment by frank-zsy** [comment]
@tyn1998 Great idea of the latter one, impressive! Looking forward to the new version!

**Comment by frank-zsy** [comment]
The data is here: https://oss.x-lab.info/repo_detail/hypertrons/hypertrons-crx.json, replace owner and repo to get data for different repos.

**Comment by tyn1998** [comment]
Hi guys, I have come up with a more clear design:

<img width="1336" alt="image" src="https://user-images.githubusercontent.com/32434520/189128041-b12eb898-f453-4d7e-b8dc-41583d6042f7.png">

There is no need to present them all together in another single area I think, so I will close #431.

**Comment by wxharry** [comment]
I think we need to think about the way to trigger those diagrams. Users will frequently click `issues`, `pull requests` and it could be annoying if the diagram is triggered each time they click it.

**Comment by tyn1998** [comment]
Yeah, there will be a delay after mouse hovering on those elements. Between 500ms and 1000ms maybe.

**Comment by tyn1998** [comment]
The fluent-ui control [TeachingBubble](https://developer.microsoft.com/en-us/fluentui#/controls/web/teachingbubble) is not flexible:

<img width="930" alt="image" src="https://user-images.githubusercontent.com/32434520/189682299-d37657d9-c759-4062-ae05-574f4c08a37c.png">

Fortunately we have [react-tooltip](https://www.npmjs.com/package/react-tooltip), which is way more customizable:

<img width="878" alt="image" src="https://user-images.githubusercontent.com/32434520/189682688-7421478c-5696-4840-9d88-a7259fff72e7.png">

**Comment by tyn1998** [comment]
Hi @frank-zsy, I have encountered  many data missing? Is that expected?

<img width="1057" alt="image" src="https://user-images.githubusercontent.com/32434520/193964258-490153d6-0956-4de5-a78d-e2b95c3a4903.png">

**Comment by frank-zsy** [comment]
@tyn1998 I am not sure why I use lowercase for all the repo names, so the file is at `https://oss.x-lab.info/repo_detail/microsoft/typescript.json` , I will fix this in recent update.
